packages/react/src/search/useSearch.ts

```diff
@@ -0,0 +1,196 @@
+   */
+  totalPages: number;
+  /**
+   * The error object if the last search request failed, or null if no error occurred.
+   */
+  error: Error | null;
+}
+
+/**
+ * React hook for performing search queries with pagination.
+ * @param {UseSearchOptions} options - Configuration options for the search hook.
+ * @returns {UseSearchState} The search state.
+ * @public
+ */
+export const useSearch = <T = GenericFields>(
+  options: UseSearchOptions<T>
+): UseSearchState<T> => {
+  const { query, page = DEFAULT_PAGE, searchIndexId, pageSize = DEFAULT_PAGE_SIZE, sort } = options;
+
+  if (!searchIndexId) {
+    throw new Error('useSearch: searchIndexId is required');
+  }
+
+  const [state, setState] = useState<{
+    results: T[];
+    isLoading: boolean;
+    isSuccess: boolean;
+    isError: boolean;
+    total: number;
+    totalPages: number;
+    error: Error | null;
+  }>({
+    results: [],
+    isLoading: true,
+    isSuccess: false,
+    isError: false,
+    total: 0,
+    totalPages: 0,
+    error: null,
+  });
+
+  const searchService = useSearchService();
+  const abortControllerRef = useRef<AbortController | null>(null);
+
+  const search = useCallback(async () => {
+    if (abortControllerRef.current) {
+      abortControllerRef.current.abort();
+    }
+
+    abortControllerRef.current = new AbortController();
+    const signal = abortControllerRef.current.signal;
+
+    setState({
+      results: [],
+      isLoading: true,
+      isSuccess: false,
+      isError: false,
+      total: 0,
+      totalPages: 0,
+      error: null,
+    });
+
+    try {
+      const offset = getOffset(page, pageSize);
+
+      const searchParams: SearchParameters<T> = {
+        searchIndexId,
+        keyphrase: query,
+        limit: pageSize,
+        offset,
+        sort,
+      };
+
+      const { results: searchResults, total } = await searchService.search<T>(searchParams, {
+        signal,
+      });
+
+      if (signal.aborted) {
+        return;
+      }
+
+      const totalPages = Math.ceil(total / pageSize);
+
+      setState({
+        results: searchResults,
+        total,
+        totalPages,
+        isSuccess: true,
+        isError: false,
+        isLoading: false,
+        error: null,
+      });
+    } catch (err) {
+      // Don't set error if request was aborted
+      if (signal.aborted) {
+        return;
+      }
+
+      const errorMessage = err instanceof Error ? err : new Error('Search failed');
+      setState({
+        results: [],
+        total: 0,
+        totalPages: 0,
+        isSuccess: false,
+        isError: true,
+        isLoading: false,
+        error: errorMessage,
+      });
+    }
+  }, [searchService, searchIndexId, page, pageSize, sort, query]);
+
+  useEffect(() => {
+    search();
+  }, [search]);
+
+  useEffect(() => {
+    return () => {
+      // Abort all requests when component unmounts
+      abortControllerRef.current?.abort();
+    };
+  }, []);
+
+  return useMemo(
```

Comment:
Currently this useMemo just returns all the same properties you have in useState, that is excessive, as state is referentially stable. Memo might be useful here if you do any transformation